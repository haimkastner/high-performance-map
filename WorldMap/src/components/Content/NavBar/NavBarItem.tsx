import { createStyles, Grid, makeStyles, Theme, Typography } from '@material-ui/core';
import React from 'react';

interface Props {
  children?: JSX.Element;
  width: number | string;
  height: number | string;
  name: string;
  value: number | string;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    navItemInput: {
      backgroundColor: '#1D1D1D',
      borderColor: 'rgba(255, 255, 255, 0.5)',
      borderStyle: 'solid',
      borderWidth: 1,
      boxSizing: 'border-box',
      color: '#FFFFFF',
      fontFamily: 'Assistant',
      opacity: 0.6,
      textAlign: 'center',
    },
    navItemName: {
      color: '#FFFFFF',
      fontFamily: 'Assistant',
      fontSize: 16,
      fontWeight: 600,
    },
  }),
);

export function NavBarItem(props: Props) {
  const classes = useStyles();

  const navItemStyle = {
    height: props.height,
    width: props.width,
  };

  return (
    <Grid container direction="row" justify="space-around" alignItems="center" style={navItemStyle}>
      <Typography id="name" className={classes.navItemName}>
        {props.name}
      </Typography>
      <Typography id="value" className={classes.navItemInput} style={navItemStyle}>
        {props.value.toString()}
      </Typography>
    </Grid>
  );
}
