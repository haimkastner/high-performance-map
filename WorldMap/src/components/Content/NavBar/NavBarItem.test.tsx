import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { NavBarItem } from './NavBarItem';

describe('NavBarItem Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NavBarItem name={'nav-bar-test'} value={4} width={3} height={3} />, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<NavBarItem name={'nav-bar-test'} value={4} width={3} height={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('NavBarItem Component Logic', () => {
    it('should render the correct name and value', () => {
      const name = 'Igal';
      const value = 78;

      const wrapper = shallow(<NavBarItem name={name} value={value} width={3} height={3} />);
      const nameP = wrapper.find('#name');
      const valueP = wrapper.find('#value');

      const nameText = nameP.text();
      const valueText = valueP.text();
      expect(nameText).toEqual(name);
      expect(valueText).toEqual(value.toString());
    });
  });
});
