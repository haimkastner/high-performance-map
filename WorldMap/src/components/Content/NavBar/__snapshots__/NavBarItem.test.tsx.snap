// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarItem Component matches the snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
  style={
    Object {
      "height": 3,
      "width": 3,
    }
  }
>
  <p
    className="MuiTypography-root makeStyles-navItemName-137 MuiTypography-body1"
    id="name"
  >
    nav-bar-test
  </p>
  <p
    className="MuiTypography-root makeStyles-navItemInput-136 MuiTypography-body1"
    id="value"
    style={
      Object {
        "height": 3,
        "width": 3,
      }
    }
  >
    4
  </p>
</div>
`;
